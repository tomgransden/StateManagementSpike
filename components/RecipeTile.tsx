import React from 'react';
import {
  TouchableOpacity,
  Image,
  View,
  Text,
  ViewStyle,
  TextStyle,
  Dimensions,
  StyleSheet,
  ImageStyle,
} from 'react-native';
import {PublicRecipeSearchResult} from '../types/publicRecipeTypes';

interface RecipeTileProps {
  item: PublicRecipeSearchResult;
  onPress: () => void;
}

const screenWidth = Dimensions.get('screen').width;

const RecipeTile = ({item, onPress}: RecipeTileProps) => (
  <TouchableOpacity style={style.tileContainer} onPress={onPress}>
    <Image
      source={{
        uri: `${item.image.src}&width=${screenWidth}&height=${screenWidth}`,
      }}
      style={style.image}
    />
    <View style={style.textContainer}>
      <Text style={style.text}>{item.title}</Text>
    </View>
  </TouchableOpacity>
);

type RecipeTileStyle = {
  tileContainer: ViewStyle;
  image: ImageStyle;
  textContainer: ViewStyle;
  text: TextStyle;
};

const style = StyleSheet.create<RecipeTileStyle>({
  tileContainer: {height: screenWidth},
  image: {width: screenWidth, height: screenWidth},
  textContainer: {
    position: 'absolute',
    bottom: 30,
    left: 0,
    right: 0,
    backgroundColor: 'rgba(0,0,0,0.7)',
    minHeight: 60,
    justifyContent: 'center',
    paddingHorizontal: 12,
  },
  text: {
    color: '#fffefe',
    textAlign: 'center',
    fontSize: 24,
  },
});

export default RecipeTile;
