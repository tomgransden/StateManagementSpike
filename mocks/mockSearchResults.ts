const mockSearchResults = {
  data: [
    {
      image: {
        src: 'https://asset.slimmingworld.co.uk/content/media/16967/spaghetti-and-meatballs_sw_recipe.jpg?v1=JGXiore20qg9NNIj0tmc3TKfKw-jr0s127JqqpCA2x7sMviNgcAYh1epuS_Lqxebn9V_qusKHfwbF7MOUrAPptzBhXIUL1Xnq2Mmdvx4fOk',
        alt: '',
        focalPoint: null,
        caption: '',
        _type: 'Image',
        id: '29361',
      },
      characteristics: 193,
      palette: 'white',
      servings: 4,
      servingType: 'per person',
      syns: 0.0,
      totalTime: 45,
      isFoodRange: false,
      foodRangeCategoryId: 0,
      recipeDbId: '1096758843477585920',
      introduction: null,
      id: '29360',
      link: '/recipes/spaghetti-and-meatballs',
      title: 'Spaghetti and meatballs',
      _type: 'RecipeTile',
    },
    {
      image: {
        src: 'https://asset.slimmingworld.co.uk/content/media/16888/pizza-pittas_sw_recipe.jpg?v1=JGXiore20qg9NNIj0tmc3TKfKw-jr0s127JqqpCA2x7sMviNgcAYh1epuS_Lqxebn9V_qusKHfwbF7MOUrAPptzBhXIUL1Xnq2Mmdvx4fOk',
        alt: '',
        focalPoint: null,
        caption: '',
        _type: 'Image',
        id: '29260',
      },
      characteristics: 448,
      palette: 'white',
      servings: 1,
      servingType: 'per person',
      syns: 13.0,
      totalTime: 20,
      isFoodRange: false,
      foodRangeCategoryId: 0,
      recipeDbId: '1090036649519214592',
      introduction: null,
      id: '29259',
      link: '/recipes/pizza-pitta',
      title: 'Pizza pitta ',
      _type: 'RecipeTile',
    },
    {
      image: {
        src: 'https://asset.slimmingworld.co.uk/content/media/16838/apricot-coconut-and-lime-balls_sw_recipe.jpg?v1=JGXiore20qg9NNIj0tmc3TKfKw-jr0s127JqqpCA2x7sMviNgcAYh1epuS_Lqxebn9V_qusKHfwbF7MOUrAPptzBhXIUL1Xnq2Mmdvx4fOk',
        alt: '',
        focalPoint: null,
        caption: '',
        _type: 'Image',
        id: '29169',
      },
      characteristics: 2832,
      palette: 'white',
      servings: 8,
      servingType: 'each',
      syns: 7.0,
      totalTime: 10,
      isFoodRange: false,
      foodRangeCategoryId: 0,
      recipeDbId: '1081945670534819840',
      introduction: null,
      id: '29168',
      link: '/recipes/apricot-coconut-and-lime-balls',
      title: 'Apricot, coconut and lime balls',
      _type: 'RecipeTile',
    },
    {
      image: {
        src: 'https://asset.slimmingworld.co.uk/content/media/16792/slow-cooker-chicken-casserole_sw_recipe.jpg?v1=JGXiore20qg9NNIj0tmc3TKfKw-jr0s127JqqpCA2x7sMviNgcAYh1epuS_Lqxebn9V_qusKHfwbF7MOUrAPptzBhXIUL1Xnq2Mmdvx4fOk',
        alt: '',
        focalPoint: null,
        caption: '',
        _type: 'Image',
        id: '29118',
      },
      characteristics: 514,
      palette: 'white',
      servings: 8,
      servingType: 'per person',
      syns: 0.0,
      totalTime: 210,
      isFoodRange: false,
      foodRangeCategoryId: 0,
      recipeDbId: '1077597437897469952',
      introduction: null,
      id: '29117',
      link: '/recipes/slow-cooker-chicken-casserole',
      title: 'Slow-cooker chicken casserole',
      _type: 'RecipeTile',
    },
    {
      image: {
        src: 'https://asset.slimmingworld.co.uk/content/media/16791/salmon-and-black-bean-taco-style-wraps_sw_recipe.jpg?v1=JGXiore20qg9NNIj0tmc3TKfKw-jr0s127JqqpCA2x7sMviNgcAYh1epuS_Lqxebn9V_qusKHfwbF7MOUrAPptzBhXIUL1Xnq2Mmdvx4fOk',
        alt: '',
        focalPoint: null,
        caption: '',
        _type: 'Image',
        id: '29120',
      },
      characteristics: 100,
      palette: 'white',
      servings: 4,
      servingType: 'per person',
      syns: 0.0,
      totalTime: 30,
      isFoodRange: false,
      foodRangeCategoryId: 0,
      recipeDbId: '1077592235811270656',
      introduction: null,
      id: '29119',
      link: '/recipes/salmon-and-black-bean-taco-style-wraps',
      title: 'Salmon and black bean taco-style wraps',
      _type: 'RecipeTile',
    },
    {
      image: {
        src: 'https://asset.slimmingworld.co.uk/content/media/16790/piri-piri-beef-noodles_sw_recipe.jpg?v1=JGXiore20qg9NNIj0tmc3TKfKw-jr0s127JqqpCA2x7sMviNgcAYh1epuS_Lqxebn9V_qusKHfwbF7MOUrAPptzBhXIUL1Xnq2Mmdvx4fOk',
        alt: '',
        focalPoint: null,
        caption: '',
        _type: 'Image',
        id: '29122',
      },
      characteristics: 129,
      palette: 'white',
      servings: 4,
      servingType: 'per person',
      syns: 0.0,
      totalTime: 20,
      isFoodRange: false,
      foodRangeCategoryId: 0,
      recipeDbId: '1077575960951255040',
      introduction: null,
      id: '29121',
      link: '/recipes/piri-piri-beef-noodles',
      title: 'Piri-piri beef noodles',
      _type: 'RecipeTile',
    },
    {
      image: {
        src: 'https://asset.slimmingworld.co.uk/content/media/16789/coffee-and-banana-pots_sw_recipe.jpg?v1=JGXiore20qg9NNIj0tmc3TKfKw-jr0s127JqqpCA2x7sMviNgcAYh1epuS_Lqxebn9V_qusKHfwbF7MOUrAPptzBhXIUL1Xnq2Mmdvx4fOk',
        alt: '',
        focalPoint: null,
        caption: '',
        _type: 'Image',
        id: '29124',
      },
      characteristics: 320,
      palette: 'white',
      servings: 4,
      servingType: 'per person',
      syns: 0.0,
      totalTime: 5,
      isFoodRange: false,
      foodRangeCategoryId: 0,
      recipeDbId: '1077553476159926272',
      introduction: null,
      id: '29123',
      link: '/recipes/coffee-and-banana-breakfast-pots',
      title: 'Coffee and banana breakfast pots',
      _type: 'RecipeTile',
    },
    {
      image: {
        src: 'https://asset.slimmingworld.co.uk/content/media/16788/massaman-curry_sw_recipe.jpg?v1=JGXiore20qg9NNIj0tmc3TKfKw-jr0s127JqqpCA2x7sMviNgcAYh1epuS_Lqxebn9V_qusKHfwbF7MOUrAPptzBhXIUL1Xnq2Mmdvx4fOk',
        alt: '',
        focalPoint: null,
        caption: '',
        _type: 'Image',
        id: '29126',
      },
      characteristics: 2304,
      palette: 'white',
      servings: 4,
      servingType: 'per person',
      syns: 0.0,
      totalTime: 25,
      isFoodRange: false,
      foodRangeCategoryId: 0,
      recipeDbId: '1077556113232097280',
      introduction: null,
      id: '29125',
      link: '/recipes/massaman-curry',
      title: 'Massaman curry',
      _type: 'RecipeTile',
    },
    {
      image: {
        src: 'https://asset.slimmingworld.co.uk/content/media/16727/air-fryer-pop-tart_sw_recipe.jpg?v1=JGXiore20qg9NNIj0tmc3TKfKw-jr0s127JqqpCA2x7sMviNgcAYh1epuS_Lqxebn9V_qusKHfwbF7MOUrAPptzBhXIUL1Xnq2Mmdvx4fOk',
        alt: '',
        focalPoint: null,
        caption: '',
        _type: 'Image',
        id: '29005',
      },
      characteristics: 480,
      palette: 'white',
      servings: 1,
      servingType: 'per person',
      syns: 8.0,
      totalTime: 10,
      isFoodRange: false,
      foodRangeCategoryId: 0,
      recipeDbId: '1070485686517231616',
      introduction: null,
      id: '29004',
      link: '/recipes/air-fryer-berry-pop-tart',
      title: 'Air fryer berry pop tart',
      _type: 'RecipeTile',
    },
    {
      image: {
        src: 'https://asset.slimmingworld.co.uk/content/media/16733/air-fryer-roast-chicken_sw_recipe.jpg?v1=JGXiore20qg9NNIj0tmc3TKfKw-jr0s127JqqpCA2x7sMviNgcAYh1epuS_Lqxebn9V_qusKHfwbF7MOUrAPptzBhXIUL1Xnq2Mmdvx4fOk',
        alt: '',
        focalPoint: null,
        caption: '',
        _type: 'Image',
        id: '29011',
      },
      characteristics: 194,
      palette: 'white',
      servings: 2,
      servingType: 'per person',
      syns: 4.0,
      totalTime: 35,
      isFoodRange: false,
      foodRangeCategoryId: 0,
      recipeDbId: '1070474765841461248',
      introduction: null,
      id: '29010',
      link: '/recipes/air-fryer-roast-chicken-dinner',
      title: 'Air fryer roast chicken dinner',
      _type: 'RecipeTile',
    },
    {
      image: {
        src: 'https://asset.slimmingworld.co.uk/content/media/16721/air-fryer-falafel_sw_recipe.jpg?v1=JGXiore20qg9NNIj0tmc3TKfKw-jr0s127JqqpCA2x7sMviNgcAYh1epuS_Lqxebn9V_qusKHfwbF7MOUrAPptzBhXIUL1Xnq2Mmdvx4fOk',
        alt: '',
        focalPoint: null,
        caption: '',
        _type: 'Image',
        id: '28995',
      },
      characteristics: 2432,
      palette: 'white',
      servings: 2,
      servingType: 'per person',
      syns: 7.0,
      totalTime: 30,
      isFoodRange: false,
      foodRangeCategoryId: 0,
      recipeDbId: '1070458406701101056',
      introduction: null,
      id: '28994',
      link: '/recipes/lentil-and-rocket-air-fryer-falafels',
      title: 'Lentil and rocket air fryer falafels',
      _type: 'RecipeTile',
    },
    {
      image: {
        src: 'https://asset.slimmingworld.co.uk/content/media/16722/air-fryer-baked-oats_sw_recipe.jpg?v1=JGXiore20qg9NNIj0tmc3TKfKw-jr0s127JqqpCA2x7sMviNgcAYh1epuS_Lqxebn9V_qusKHfwbF7MOUrAPptzBhXIUL1Xnq2Mmdvx4fOk',
        alt: '',
        focalPoint: null,
        caption: '',
        _type: 'Image',
        id: '28997',
      },
      characteristics: 480,
      palette: 'white',
      servings: 1,
      servingType: 'per person',
      syns: 9.0,
      totalTime: 15,
      isFoodRange: false,
      foodRangeCategoryId: 0,
      recipeDbId: '1070456050445971456',
      introduction: null,
      id: '28996',
      link: '/recipes/confetti-baked-oats-air-fryer',
      title: 'Air fryer confetti baked oats ',
      _type: 'RecipeTile',
    },
  ],
  pagination: {limit: 12, offset: '0', hasMore: true, total: '185'},
};

export default mockSearchResults;
