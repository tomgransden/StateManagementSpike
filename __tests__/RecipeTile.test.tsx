import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react-native';
import RecipeTile from '../components/RecipeTile';
import mockSearchResults from '../mocks/mockSearchResults';

describe('RecipeTile', () => {
  it('Should render correctly', () => {
    render(<RecipeTile item={mockSearchResults.data[0]} onPress={() => {}} />);

    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('Should call onPress', () => {
    const mockOnPress = jest.fn();
    render(
      <RecipeTile item={mockSearchResults.data[0]} onPress={mockOnPress} />,
    );

    fireEvent.press(screen.getByText('Spaghetti and meatballs'));

    expect(mockOnPress).toBeCalled();
  });
});
