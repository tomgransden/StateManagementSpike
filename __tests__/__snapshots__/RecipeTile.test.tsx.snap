// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeTile Should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 750,
      "opacity": 1,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://asset.slimmingworld.co.uk/content/media/16967/spaghetti-and-meatballs_sw_recipe.jpg?v1=JGXiore20qg9NNIj0tmc3TKfKw-jr0s127JqqpCA2x7sMviNgcAYh1epuS_Lqxebn9V_qusKHfwbF7MOUrAPptzBhXIUL1Xnq2Mmdvx4fOk&width=750&height=750",
      }
    }
    style={
      Object {
        "height": 750,
        "width": 750,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.7)",
        "bottom": 30,
        "justifyContent": "center",
        "left": 0,
        "minHeight": 60,
        "paddingHorizontal": 12,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fffefe",
          "fontSize": 24,
          "textAlign": "center",
        }
      }
    >
      Spaghetti and meatballs
    </Text>
  </View>
</View>
`;
